engine/internal/s3/client.go

```diff
@@ -56,3 +56,31 @@ func (c *Client) Download(ctx context.Context, w io.WriterAt, key string) error
 	}
 	return nil
 }
+
+
+// ListObjectsPages returns S3 objects with pagination.
+func (c *Client) ListObjectsPages(
+	ctx context.Context,
+	prefix string,
+	f func(page *s3.ListObjectsV2Output, lastPage bool) bool,
+) error {
+	const maxKeys = int32(1000)
+	p := s3.NewListObjectsV2Paginator(
+		c.svc,
+		&s3.ListObjectsV2Input{
+			Bucket: aws.String(c.bucket),
+			Prefix: aws.String(prefix),
+		},
+		func(o *s3.ListObjectsV2PaginatorOptions) {
```

Comment:
This looks unnecessary since the default limit is 1000.

https://pkg.go.dev/github.com/aws/aws-sdk-go-v2/service/s3#ListObjectsV2PaginatorOptions

> By default, the action returns up to 1,000 key name